dev-lang/python/python-3.11.0_beta3.ebuild

```diff
@@ -203,6 +204,44 @@ src_configure() {
 	local -x LDFLAGS_NODIST=${LDFLAGS}
 	local -x CFLAGS= LDFLAGS=
 
+	if tc-is-cross-compiler ; then
+		# We need to build our own Python on CBUILD first, and feed it in.
+		# bug #847910
+		local myeconfargs_cbuild=(
+			"${myeconfargs[@]}"
+
+			# As minimal as possible for the mini CBUILD Python
+			# we build just for cross, to satisfy --with-build-python.
+			--without-lto
+			--without-readline
+			--disable-optimizations
+		)
+
+		myeconfargs+=(
+			# Point the imminent CHOST build to the Python we just
+			# built for CBUILD.
+			--with-build-python="${WORKDIR}"/${P}-${CBUILD}/python
+		)
+
+		mkdir "${WORKDIR}"/${P}-${CBUILD} || die
+		pushd "${WORKDIR}"/${P}-${CBUILD} &> /dev/null || die
+		ECONF_SOURCE="${S}" econf_build "${myeconfargs_cbuild[@]}"
+
+		# Avoid as many dependencies as possible for the cross build.
+		local mod
+		for mod in NIS _DBM _GDBM _DBM _SQLITE3 _HASHLIB _SSL _CURSES \
+			_CURSES_PANEL READLINE _TKINTER PYEXPAT ; do
+			echo "MODULE_${mod}=disabled" >> Makefile || die
+		done
```

Comment:
Perhaps just inline the full list with `<<-EOF`?

Reply:
Sure.